rust/src/operations/merge.rs

```diff
@@ -0,0 +1,1614 @@
+//! Merge data from a source dataset with the target Delta Table based on a join
+//! predicate.  A full outer join is performed which results in source and
+//! target records that match, source records that do not match, or target
+//! records that do not match.
+//!
+//! Users can specify update, delete, and insert operations for these categories
+//! and specify additional predicates for finer control. The order of operations
+//! specified matter.  See [`MergeBuilder`] for more information
+//!
+//! *WARNING* The current implementation rewrites the entire delta table so only
+//! use on small to medium sized tables. Enhancements tracked at #850
```

Comment:
Also worth noting that it currently loads the entire thing into memory and uses only 1 thread, IIUC.